sdk/cosmos/azure-cosmos/azure/cosmos/aio/scripts.py

```diff
@@ -189,31 +199,34 @@ async def delete_stored_procedure(self, sproc, **kwargs):
     @distributed_trace_async
     async def execute_stored_procedure(
         self,
-        sproc,  # type: Union[str, Dict[str, Any]]
-        partition_key=None,  # type: Optional[str]
-        params=None,  # type: Optional[List[Any]]
-        enable_script_logging=None,  # type: Optional[bool]
-        **kwargs  # type: Any
-    ):
-        # type: (...) -> Any
+        sproc: Union[str, Dict[str, Any]],
+        partition_key: Optional[str] = None,
+        params: Optional[List[Any]] = None,
+        enable_script_logging: Optional[bool] = None,
+        **kwargs: Any
+    ) -> Dict[str, Any]:
         """Execute a specified stored procedure.
 
         If the stored procedure does not already exist in the container, an exception is raised.
 
         :param sproc: The ID (name) or dict representing the stored procedure to be executed.
+        :type sproc: str or Dict[str, Any]
         :param partition_key: Specifies the partition key to indicate which partition the sproc should execute on.
+        :type partition_key: Optional[str]
         :param params: List of parameters to be passed to the stored procedure to be executed.
-        :param bool enable_script_logging: Enables or disables script logging for the current request.
-        :returns: Result of the executed stored procedure for the given parameters.
+        :type params: Optional[List[Any]]
+        :param enable_script_logging: Enables or disables script logging for the current request.
+        :type enable_script_logging: Optional[bool]
         :raises ~azure.cosmos.exceptions.CosmosHttpResponseError: If the stored procedure execution failed
             or if the stored procedure with given id does not exists in the container.
-        :rtype: dict[str, Any]
+        :returns: Result of the executed stored procedure for the given parameters.
+        :rtype: Dict[str, Any]
         """
 
         request_options = _build_options(kwargs)
         if partition_key is not None:
             request_options["partitionKey"] = (
-                _cosmos_client_connection._return_undefined_or_empty_partition_key(
+                _CosmosClientConnection._return_undefined_or_empty_partition_key(
```

Comment:
Is this a staticmethod?

Reply:
_return_undefined_or_empty_partition_key is a static method in the client connection class